cmd/gazelle/fix-update.go

```diff
@@ -322,11 +323,22 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 			mappedKinds    []config.MappedKind
 			mappedKindInfo = make(map[string]rule.KindInfo)
 		)
-		for _, r := range gen {
-			if repl, ok := c.KindMap[r.Kind()]; ok {
+		// We apply map_kind to all rules, including pre-existing ones.
+		var allRules []*rule.Rule
+		allRules = append(allRules, gen...)
+		if f != nil {
+			allRules = append(allRules, f.Rules...)
+		}
+		for _, r := range allRules {
+			repl, err := lookupMapKindReplacement(c.KindMap, r.Kind())
+			if err != nil {
+				errorsFromWalk = append(errorsFromWalk, fmt.Errorf("couldn't handle loop in map_kinds: %w", err))
```

Comment:
nit: the final error would be something like "couldn't handle loop in map_kinds: found loop of map_kind replacements: xxx", which is quite redundant. You can either append the error without any wrapping, or wrap with the context here like "looking up mapped kind: found loop of map_kind replacements: xxx".